### ISSUE TITLE
MultiResolutionDetector.forward/detect accept a mask argument and silently ignore it

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4090]
`MultiResolutionDetector.forward` and `.detect` both accept a `mask` argument and neither
uses it. `KeyNetDetector` inherits both. A caller that passes a mask gets exactly the same
detections as one who does not.

## Reproduction

```python
import torch
from kornia.feature import MultiResolutionDetector
from kornia.feature.responses import BlobHessian

torch.manual_seed(0)
det = MultiResolutionDetector(BlobHessian(), num_features=100)
img = torch.rand(1, 1, 64, 64)

l1, r1 = det(img)
l2, r2 = det(img, torch.zeros(1, 1, 64, 64))   # mask everything out
print(torch.equal(l1, l2) and torch.equal(r1, r2))
```

```
True
```

An all-zero mask should suppress every detection; instead the output is bit-identical.

## Where

`kornia/feature/scale_space_detector.py:519` and `:585`. `forward` forwards `mask` to
`detect`, and `detect` never reads it.

The docstring on both methods documents it as functional:

> `mask`: a mask with weights where to apply the response function. The shape must be the
> same as the input image.

`ScaleSpaceDetector.detect` does honour its `mask` (via `_create_octave_mask`), so the two
detectors in this file disagree on the same documented argument.

## History

The argument has been inert since `MultiResolutionDetector` was introduced in #1574; the
#3623 refactor did not change it either way.

## Options

1. Implement it — multiply the response map by a resized mask in
   `detect_features_on_single_level`, matching `ScaleSpaceDetector`'s
   `_create_octave_mask` behaviour. Behaviour-changing for anyone currently passing a mask
   and (silently) getting unmasked results.
2. Raise `NotImplementedError` when `mask is not None` and delete the docstring paragraph.
3. Drop the parameter — breaks the shared `(img, mask)` call signature with
   `ScaleSpaceDetector`.

Option 1 matches what the docstring already promises. Reproduced on `main` at 9abf06d9,
torch 2.9.1, CPU.
